pkg/volume/azure_file/azure_file.go

```diff
@@ -208,15 +209,24 @@ func (b *azureFileMounter) SetUpAt(dir string, fsGroup *int64) error {
 	if accountName, accountKey, err = b.util.GetAzureCredentials(b.plugin.host, b.pod.Namespace, b.secretName); err != nil {
 		return err
 	}
-	os.MkdirAll(dir, 0700)
 
-	source := fmt.Sprintf("//%s.file.%s/%s", accountName, getStorageEndpointSuffix(b.plugin.host.GetCloudProvider()), b.shareName)
-	// parameters suggested by https://azure.microsoft.com/en-us/documentation/articles/storage-how-to-use-files-linux/
-	options := []string{fmt.Sprintf("vers=3.0,username=%s,password=%s,dir_mode=0700,file_mode=0700", accountName, accountKey)}
-	if b.readOnly {
-		options = append(options, "ro")
+	mountOptions := []string{}
+	source := ""
+
+	if runtime.GOOS == "windows" {
+		source = fmt.Sprintf("\\\\%s.file.%s\\%s", accountName, getStorageEndpointSuffix(b.plugin.host.GetCloudProvider()), b.shareName)
+		mountOptions = []string{"use", "*", source, fmt.Sprintf("/u:AZURE\\%s", accountName), accountKey, "/p:yes"}
+	} else {
+		os.MkdirAll(dir, 0700)
```

Comment:
Hi, I did not change the original value 0700, is there any reason to change it to 0755. Seems 0700 is more safe as it's only accessible by root user.